lightkurve/targetpixelfile.py

```diff
@@ -684,6 +684,216 @@ def update(cadence, log_stretch):
+                                               ("Time (ISO)", "@time_iso"),
+                                               ("Flux", "@flux"),
+                                               ("Quality Code", "@quality_code"),
+                                               ("Quality Flag", "@quality")],
+                                     renderers=[r],
+                                     mode='mouse',
+                                     point_policy="snap_to_data"))
+            # Vertical line to indicate the cadence shown in Fig 2
+            vert = Span(location=0, dimension='height', line_color='firebrick',
+                        line_width=4, line_alpha=0.5)
+            fig1.add_layout(vert)
+
+
+            fig2 = figure(plot_width=300, plot_height=300, x_range=(min(xx), max(xx+1)),
+                        y_range=(min(yy), max(yy+1)), title='Target Pixel File',
+                        tools='tap, box_select, wheel_zoom, reset')
+            fig2.yaxis.axis_label = 'Pixel Row Number'
+            fig2.xaxis.axis_label = 'Pixel Column Number'
+
+            pedestal = np.nanmin(self.flux)
+            vlo, lo, med, hi, vhi = np.nanpercentile(self.flux-pedestal, [0.2, 1, 50, 95, 99.8])
+            vstep = (np.log10(vhi) - np.log10(vlo)) / 300.0  # assumes counts >> 1.0!
+            color_mapper = LogColorMapper(palette="Viridis256", low=lo, high=hi)
+
+            fig2_dat = fig2.image([pedestal+self.flux[0,:,:]], x=self.column, y=self.row,
+                              dw=self.shape[2], dh=self.shape[1], dilate=True,
+                              color_mapper=color_mapper)
+
+            preselection = Selection()
+            preselection.indices=pixel_index_array[self.pipeline_mask].reshape(-1).tolist()
+            source2 = ColumnDataSource(data=dict(xx=xa+0.5, yy=ya+0.5), selected=preselection)
+            r1 = fig2.rect('xx', 'yy', 1, 1, source=source2, fill_color='gray', fill_alpha=0.4, line_color='white')
+
+            def callback(attr, old, new):
+                if len(source2.selected.indices) > 0:
+                    selected_indices = np.array(source2.selected.indices)
+                    selected_mask = np.isin(pixel_index_array, selected_indices)
+                    lc_new = self.to_lightcurve(aperture_mask=selected_mask)
+                    source.data = dict(time=lc.time, flux=lc_new.flux,
+                                    cadence=lc.cadenceno, quality=lc.quality)
+                else:
+                    source.data = dict(time=lc.time, flux=lc.flux*0.0,
+                                    cadence=lc.cadenceno, quality=lc.quality)
+                callback4('junk')
+
+            source2.on_change('selected', callback)
+
+            n_cadences = len(self.time)
+            amp_slider = Slider(start=0, end=n_cadences-1, value=0, step=1,
+                    title="TPF slice index", width = 800)
+
+            def callback2(attr, old, new):
```

Comment:
You'll want to give the various callback functions a more descriptive name, e.g. `callback2` could be `cadence_slider_callback`.